### PR TITLE
BDT-101095

### PR DESCRIPTION
**Pull request by mrfreeman**

### CODE REVIEW

File: services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/AzureKeyVaultTokenService.java [hunk 1]
```diff
@@ -20,22 +19,30 @@ public class AzureKeyVaultTokenService implements KeyVaultTokenService {
...
-                .tenantId(globalParamsMap.get(GlobalParams.TENANT_ID.getValue()))
-                .username(globalParamsMap.get(GlobalParams.USERNAME.getValue()))
-                .password(globalParamsMap.get(GlobalParams.PASSWORD.getValue()))
-                .build();
+        if (!globalParamsMap.get(GlobalParams.AUTHENTICATION_METHOD.getValue()).equals(GlobalParams.CLIENT_CREDENTIALS.getValue())) {
+            throw new IllegalArgumentException("Authentication method not supported");
+        }
+
+        TokenCredential credential = getTokenCredential(globalParamsMap);
+//                : new ClientCertificateCredentialBuilder() // TODO: implement if needed
```
Comment (jacob-yeger) [inline services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/AzureKeyVaultTokenService.java:27]: lets remove commented out code

File: services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/AzureKeyVaultTokenService.java [hunk 2]
```diff
@@ -20,22 +19,30 @@ public class AzureKeyVaultTokenService implements KeyVaultTokenService {
     public AccessToken fetchAccessToken(ExecutionContext executionContext) {
         Map<String, String> globalParamsMap = ParamsMapUtils.getGlobalParamsMap(executionContext);
 
-        TokenCredential credential = globalParamsMap.get(GlobalParams.AUTHENTICATION_METHOD.getValue()).equals(GlobalParams.CLIENT_SECRET.getValue())
-                ? new ClientSecretCredentialBuilder()
-                .clientId(globalParamsMap.get(GlobalParams.CLIENT_ID.getValue()))
```
Comment (jacob-yeger) [inline services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/AzureKeyVaultTokenService.java:25]: do we validate that clientId + clientSecret values exist?

**Review by jacob-yeger — APPROVED** [review]

File: services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/AzureKeyVaultTokenService.java (hunk 2 above)
Comment (mrfreeman) [reply services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/AzureKeyVaultTokenService.java:25]: > do we validate that clientId + clientSecret values exist?

sdk will throw IllegalArgumentException and we catch it later and return BAD_REQUEST

File: services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/AzureKeyVaultTokenService.java (hunk 1 above)
Comment (mrfreeman) [reply services/bigid-azure-key-vault-integration-app/src/main/java/com/bigid/azurekeyvaultapp/service/impl/AzureKeyVaultTokenService.java]: > lets remove commented out code

done

**Pull request merged**